### PR TITLE
executor: remove unnecessary APIs from `Enter`.

### PR DESCRIPTION
**Pull request by carllerche**
These APIs have never been used.

### CODE REVIEW

**Comment by carllerche** [comment]
CI passed [here](https://dev.azure.com/tokio-rs/Tokio/_build/results?buildId=769) but is not being reported.

File: tokio-executor/src/enter.rs [hunk 1]
```diff
@@ -10,8 +11,7 @@ thread_local!(static ENTERED: Cell<bool> = Cell::new(false));
 ///
 /// For more details, see [`enter` documentation](fn.enter.html)
 pub struct Enter {
-    on_exit: Vec<Box<dyn Callback>>,
-    permanent: bool,
+    _p: PhantomData<RefCell<()>>,
```
Comment (doitian) [inline tokio-executor/src/enter.rs]: Sorry to bother. I'm reading the source code recently and I'm really curious about the usage of `PhantomData` here. After some research, it looks like a trick to make `Enter` `Send` but not `Sync`. If so, why not explicitly declare it as `!Sync`. Is there any historical reason or did I miss some other reasons why `PhantomData<RefCell<()>>` is used here?

Comment (carllerche) [reply tokio-executor/src/enter.rs]: Probably just habit.

Comment (doitian) [reply tokio-executor/src/enter.rs]: Thanks for your reply. I have figured out another cause, negative trait bounds are not yet fully implemented, we can only use marker types to mark a type that is not `Send` or `Sync`.

**Pull request merged**